api/metadata/models.py

```diff
@@ -6,22 +6,20 @@
 from django.db import models
 
 from organisations.models import Organisation
-from projects.models import Project
 
 from .fields import GenericObjectID
 
 FIELD_VALUE_MAX_LENGTH = 2000
 
-METADATA_SUPPORTED_MODELS = ["environment"]
+METADATA_SUPPORTED_MODELS = ["environment", "feature", "project", "segment"]
 
 # A map of model name to a function that takes the object id and returns the organisation_id
 SUPPORTED_REQUIREMENTS_MAPPING = {
-    "environment": {
-        "organisation": lambda org_id: org_id,
-        "project": lambda project_id: Project.objects.get(
-            id=project_id
-        ).organisation_id,
-    }
+    "environment": ["organisation", "project"],
+    "feature": ["organisation", "project"],
+    "segment": ["organisation", "project"],
+    "project": [],
```

Comment:
Why did add project and organisation here?

Reply:
We need the content type of both to create a MetadataModelFieldRequirement which requires the content type ID of the organization or the project, their respective IDs, and the model_field ID of the segment, feature, or environment.

Reply:
But that's what this code [here](https://github.com/Flagsmith/flagsmith/pull/3315/files#diff-679ebeca94a92941172d9fd3c4891c62d17e65226a8b3ed19dcc8e62f688e907R86) is doing, right?

Reply:
I think there is some confusion here...

```
need_content_type_of = list(
            chain.from_iterable(
                (key, *value) for key, value in SUPPORTED_REQUIREMENTS_MAPPING.items()
            )
        )
```
This code returns a list that includes both key and values, i.e: we don't need to add `project` and `organisation` in order to return their content type

Reply:
OK, I think it's clear now, I made the correction, thank you for the help @gagantrivedi.